crates/goose/src/agents/extension_manager.rs

```diff
@@ -113,11 +114,68 @@ impl ExtensionManager {
     // TODO IMPORTANT need to ensure this times out if the extension command is broken!
     pub async fn add_extension(&mut self, config: ExtensionConfig) -> ExtensionResult<()> {
         let sanitized_name = normalize(config.key().to_string());
+
+        /// Helper function to merge environment variables from direct envs and keychain-stored env_keys
+        async fn merge_environments(
```

Comment:
is there a reason/benefit to declare this inline in the `fn` vs somewhere else once?